test/integration/setup.js

```diff
@@ -98,13 +131,16 @@ const createHTTPServer = async (path) => {
   });
 };
 
-async function runExtensionTest(t, server, driver, extensionDirName) {
+async function runExtensionTest(t, server, driver, extensionDirName, browser) {
   try {
     const url = `http://localhost:${server.address().port}`;
     const userAgent = await driver.executeScript(() => window.navigator.userAgent);
 
-    t.pass(`Connected to browser: ${userAgent}"`);
-
+    t.pass(`Connected to ${browser} browser: ${userAgent}"`);
+    if (browser === "edge") {
+      // Edge seems to need a little latency to be able to start the background page.
+      await driver.sleep(500);
```

Comment:
This is prone to time-outs. Is it feasible to notify the test runner when the background page is ready? E.g. by sending a request on start-up of the background page.